### PR TITLE
Feat: delete and publish list object

### PR DESCRIPTION
**Pull request by ristianaditya**
## Changes

- change sidebar and header component
- add object dummy response
- add limit per-page select
- refactor reusable datatable
- add delete and publish component

## Preview
image
image

## Evidence
title: add component delete and publish list object
project: Jabar Super Apps
participants: @rayfajars @johndy2742 @ristianaditya @MNAlfiansyah

### CODE REVIEW

File: dashboard/common/constant/object.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    label?: string,
+    sortable?:boolean
+}
+export const objectHeaders: IObject[] = [
+    {label: "Nama Objek", key: "name", sortable:true},
+    {label: "Pengelola", key: "organizer", sortable:true},
+    {label: "Alamat Objek", key: "address"},
+    {label: "Status", key: "status"},
+    {key: "actions"}
+]
```
Comment (Ibwedagama) [inline dashboard/common/constant/object.ts:12]: minor:

add empty line on this file @ristianaditya 👍🏼

File: dashboard/components/Base/DataTable/index.vue [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    </div>
+     <UPagination 
+     v-model="currentPage"
+     class="mt-5 mb-10 justify-end" 
+     :page-count="5" 
+     :total=total
+     :value="currentPage"
+     :prev-button="{ icon: 'i-heroicons-arrow-small-left-20-solid', label: 'Prev', color: 'gray' }" 
+     :next-button="{ icon: 'i-heroicons-arrow-small-right-20-solid', trailing: true, label: 'Next', color: 'gray' }"
+     @click="onClickPagination(currentPage)"
```
Comment (Ibwedagama) [inline dashboard/components/Base/DataTable/index.vue:69]: minor:

hello, @ristianaditya please add indentation for these props, maybe the formatted doesn't work correctly? 🤔 

cc @rayfajars

File: dashboard/components/Base/DataTable/index.vue [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+      type-modal = "warning"
+      @close="(isOpenConfirm = false)" 
+      @confirm="updateStatus()"/>
+    <UNotifications />
+  </div>
+</template>
+  
+<script setup lang="ts">
+
+    import { ref, watch } from 'vue'
```
Comment (Ibwedagama) [inline dashboard/components/Base/DataTable/index.vue:95]: minor:

doesn't Nuxt js auto import these hook from 'vue'? 🤔

File: dashboard/components/Base/DataTable/index.vue [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    }
+
+    const props = defineProps({
+        headers: {
+          type: Array as () => TableColumn[],
+          default: () => [],
+        },
+        actions: {
+          type: Object,
+          default: ()=>{}
```
Comment (Ibwedagama) [inline dashboard/components/Base/DataTable/index.vue:121]: hello @ristianaditya the default value for object should be like this:

```
default: () => ({})
```

when using arrow function the syntax look like this:

```
const arrowFunction = () => {

}
```

so when we write `()=>{}` this actually return a `void` and not an empty object 👍🏼

File: dashboard/components/Base/Modal/index.vue [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  const emit = defineEmits(['close', 'confirm']);
+  
+  function closedModal(){
+    emit('close')
+  }
+  function confirmModal(){
+    emit('confirm')
+  }
+
+</script>
```
Comment (Ibwedagama) [inline dashboard/components/Base/Modal/index.vue:121]: minor:

add empty line 👍🏼

File: dashboard/components/TheHeader.vue [hunk 6]
```diff
@@ -1,21 +1,25 @@
...
-      {{ pageTitle }}
-    </h1>
+    class="flex h-[72px] w-full items-center justify-between bg-white px-6">
+    <div class="ml-7">
+      <nav class="flex flex-wrap items-center">
+        <div class="inline-block">
+          <NuxtLink
+            class="font-roboto text-blue-gray-800 mr-4 text-md font-bold">
+            {{ activePage.page }}
+          </NuxtLink>
```
Comment (Ibwedagama) [inline dashboard/components/TheHeader.vue:11]: ask:

hello @ristianaditya this `NuxtLink` doesn't have `:to` property, is this intentional? 🤔

File: dashboard/composables/useFetchData.ts [hunk 7]
```diff
@@ -1,18 +1,148 @@
...
+export const useFetchData = async (path:string, page?: Ref<number>, search?: Ref<string>) => {
   const config = useRuntimeConfig();
   return new Promise(async (resolve, reject) => {
     try {
-      
       await useFetch(config.public.baseURL.concat(path), {
-        query: { page: page.value },
+        query: { page: page?.value, q: search },
+        headers: {
+          Authorization: `Bearer NRNALK6O8nbyXOTS7tp70AECvcKy9mwCn4hiuVOv`
```
Comment (Ibwedagama) [inline dashboard/composables/useFetchData.ts:11]: major:

hello @ristianaditya please remove this `Bearer {{token}}` from the commit, this is a best practice to never commit any credentials like token, API key, email, password, etc. use ENV_VARIABLE for static data and local storage/cookie for dynamic data like token

cc @rayfajars

File: dashboard/composables/useFetchData.ts [hunk 8]
```diff
@@ -1,18 +1,148 @@
...
+
+export const useDeleteData = async (path:string, id?: Ref<number>) => {
+  const config = useRuntimeConfig();
+  return new Promise(async (resolve, reject) => {
+    try {
+      
+      await useFetch(config.public.baseURL.concat(`${path}/${id?.value}`), {
+        method: 'delete',
+        headers: {
+          Authorization: `Bearer NRNALK6O8nbyXOTS7tp70AECvcKy9mwCn4hiuVOv`
```
Comment (Ibwedagama) [inline dashboard/composables/useFetchData.ts:138]: major:

this is also need to be changed

**Review by marsellavaleria19 — APPROVED** [review]

File: dashboard/common/constant/object.ts (hunk 1 above)
Comment (ristianaditya) [reply dashboard/common/constant/object.ts]: okay kang @Ibwedagama thanks for the advice. this is my commit: [dc015](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/dc0151c7d0990d54f23333629e02e256e64e754e)

File: dashboard/components/Base/DataTable/index.vue (hunk 3 above)
Comment (ristianaditya) [reply dashboard/components/Base/DataTable/index.vue]: okay kang @Ibwedagama thanks for the advice. this is my commit: [87d14](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/87d1478a479ba498570811898df3f5e1f6b34fb2)

File: dashboard/components/Base/DataTable/index.vue (hunk 4 above)
Comment (ristianaditya) [reply dashboard/components/Base/DataTable/index.vue]: thanks for the insight kang. I've fixed it. this is my commit : [a27fb](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/a27fbb1b02638094e75cb38bcdca1ab8b74d666d)

File: dashboard/components/Base/Modal/index.vue (hunk 5 above)
Comment (ristianaditya) [reply dashboard/components/Base/Modal/index.vue]: okay kang @Ibwedagama thanks for the advice. this is my commit: [fac41](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/fac417ffae8992a9ca01aa0a28eac5997b8b252e)

**Review by rayfajars — APPROVED** [review]

**Review by Ibwedagama — APPROVED** [review]

File: dashboard/components/TheHeader.vue (hunk 6 above)
Comment (ristianaditya) [reply dashboard/components/TheHeader.vue]: thanks for the advice. I've changed it to div. this is my commit: [bd49a](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/bd49afae94ee9868e36ea09a7258372f619904e0)

File: dashboard/composables/useFetchData.ts (hunk 7 above)
Comment (ristianaditya) [reply dashboard/composables/useFetchData.ts]: sorry for forgetting to push Authorization. but this is just a dummy token and I have deleted it. this is my commit: [dc263](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/dc2631cf39c1b6da8c7cca95d2f788b2eaad6d7d)

File: dashboard/composables/useFetchData.ts (hunk 8 above)
Comment (ristianaditya) [reply dashboard/composables/useFetchData.ts]: sorry for forgetting to push Authorization. but this is just a dummy token and I have deleted it. this is my commit: [dc263](https://github.com/jabardigitalservice/event-management-system/pull/16/commits/dc2631cf39c1b6da8c7cca95d2f788b2eaad6d7d)

**Pull request merged**